### DISCUSSION TITLE
Welcome to SpotifyAPI-NET Discussions!

### DISCUSSION THREAD

**Discussion by JohnnyCrazy** [JohnnyCrazy/SpotifyAPI-NET#545] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by absid89** [comment]
Hello, I love using this library, there is a question about the search, in my application I wanted to display new releases of artists followed by the user. But in order to realize this, it turns out that you need to select all the tracks and sort by release date every time, or not? did not find in the search method, parameters for this.

**Reply by JohnnyCrazy** [reply]
Hi @absid89 

That's correct, as to the documentation of the [search endpoint](https://developer.spotify.com/documentation/web-api/reference/search/search/):

> Currently, you cannot fetch sorted results.

**Comment by Ben2903** [comment]
Hello, how can i play an song with the API?

**Reply by JohnnyCrazy** [reply]
You can use the Player Beta API: https://developer.spotify.com/documentation/web-api/reference/player/

You can find all methods for this API here: https://github.com/JohnnyCrazy/SpotifyAPI-NET/blob/master/SpotifyAPI.Web/Clients/Interfaces/IPlayerClient.cs

```csharp
var spotify = new SpotifyClient("YourToken");

var player = spotify.Player;
await player.ResumePlayback(new PlayerResumePlaybackRequest { Uris = new List<string> { "spotify:track:XYZ" }});
```

**Comment by jeremyhallden** [comment]
Hi,

2 questions if anyone has time :)

1. Do I need to use this auth flow  - "https://github.com/JohnnyCrazy/SpotifyAPI-NET/blob/master/SpotifyAPI.Docs/docs/authorization_code.md#using-spotifywebauth" to access personal data in a short-running console app?
It seems as this flow - "https://github.com/JohnnyCrazy/SpotifyAPI-NET/blob/master/SpotifyAPI.Docs/docs/client_credentials.md" explicitly does not support user information?

2. How do I query for the special "liked songs" playlist (or get the liked status of a track some other way) using these APIs?
(https://developer.spotify.com/documentation/web-api/reference/library/get-users-saved-tracks/)

Thanks!

**Reply by JohnnyCrazy** [reply]
Hi :)

1. client_credentials does not support access to personal data, correct. For a console app, you should use [PKCE](https://johnnycrazy.github.io/SpotifyAPI-NET/docs/next/pkce)
2. That would be the "Library API" - "Get User's saved tracks" - https://developer.spotify.com/documentation/web-api/reference-beta/#category-library - https://github.com/JohnnyCrazy/SpotifyAPI-NET/blob/master/SpotifyAPI.Web/Clients/Interfaces/ILibraryClient.cs#L109 - Every song in there should be also "liked"

**Reply by jeremyhallden** [reply]
Perfect - will try that out. Thanks for the quick reply :)

**Reply by JohnnyCrazy** [reply]
One addition: It's not completely clear in the pkce docs, but you can also use Spotify.Web.Auth as a local web server

Check out the pkce example in the repo, its name is cli persistent config

**Comment by dinoginger** [comment]
Hi,

I have severe problems with finding Items via Search.Item()
A;; works fine, but the search won't work for songs with special characters for me. (As an example = search gives no results for " scream & shout") 
tried encoding query in ASCII, still the same problem. I'm kind of new to API's at all,  am I doing something wrong?

**Reply by JohnnyCrazy** [reply]
Are you sure you're correctly processing the response. The following code yields multiple (>1000) albums, tracks and playlists. The query is automatically encoded, so you should not have to do that manually

```
await spotify.Search.Item(new SearchRequest(SearchRequest.Types.All, "scream & shout"));
```

**Reply by dinoginger** [reply]
Oh, my problem was not with Search.Item(), but with query which I was passing to it. I though query must be encoded necessarily, so used HttpUtility.EncodeUrl(___, DefaultEncoding). "Scream & shout" therefore "&" was encoded to "%26", and so result returned error. Anyways, thanks for fast response 😄
